google-cloud-spanner/lib/google/cloud/spanner.rb

```diff
@@ -19,7 +19,310 @@
+    #
+    # add_users_table_sql = %q(
+    #   CREATE TABLE users (
+    #     id INT64 NOT NULL,
+    #     username STRING(25) NOT NULL,
+    #     name STRING(45) NOT NULL,
+    #     email STRING(128),
+    #   ) PRIMARY KEY(id)
+    # )
+    #
+    # database.update statements: [add_users_table_sql]
+    # ```
+    #
+    # ## Creating clients
+    #
+    # In order to read and/or write data, you must create a database client.
+    # You can think of a client as a database connection: All of your
+    # interactions with Cloud Spanner data must go through a client. Typically
+    # you create a client when your application starts up, then you re-use that
+    # client to read, write, and execute transactions.
+    #
+    # Use {Spanner::Project#client} to create a client:
+    #
+    # ```ruby
+    # require "google/cloud/spanner"
+    #
+    # spanner = Google::Cloud::Spanner.new
+    #
+    # db = spanner.client "my-instance", "my-database"
+    #
+    # results = client.execute "SELECT 1"
+    #
+    # results.rows.each do |row|
+    #   puts row
+    # end
+    # ```
+    #
+    # ## Writing data
+    #
+    # You write data using your client object. The client object supports
+    # various mutation operations, as well as combinations of inserts, updates,
+    # deletes, etc., that can be applied atomically to different rows and/or
+    # tables in a database.
+    #
+    # Use {Spanner::Client#commit} to execute various mutations atomically at a
+    # single logical point in time:
+    #
+    # ```ruby
+    # require "google/cloud/spanner"
+    #
+    # spanner = Google::Cloud::Spanner.new
+    #
+    # db = spanner.client "my-instance", "my-database"
+    #
+    # db.commit do |c|
+    #   c.update "users", [{ id: 1, name: "Charlie", active: false }]
+    #   c.insert "users", [{ id: 2, name: "Harvey",  active: true }]
+    # end
+    # ```
+    #
+    # ## Querying data using SQL
+    #
+    # Cloud Spanner supports a native SQL interface for reading data that is
+    # available through {Spanner::Client#execute}:
+    #
+    # ```ruby
+    # require "google/cloud/spanner"
+    #
+    # spanner = Google::Cloud::Spanner.new
+    #
+    # db = spanner.client "my-instance", "my-database"
+    #
+    # results = db.execute "SELECT * FROM users"
+    #
+    # results.rows.each do |row|
+    #   puts "User #{row[:id]} is #{row[:name]}"
+    # end
+    # ```
+    #
+    # ## Reading data using the read method
+    #
+    # In addition to Cloud Spanner's SQL interface, Cloud Spanner also supports
+    # a read interface. Use the {Spanner::Client#read} method to read rows from
+    # the database, and use its `keys` option to pass unique identifiers as both
+    # lists and ranges:
+    #
+    # ```ruby
+    # require "google/cloud/spanner"
+    #
+    # spanner = Google::Cloud::Spanner.new
+    #
+    # db = spanner.client "my-instance", "my-database"
+    #
+    # results = db.read "users", ["id", "name"], keys: 1..5
```

Comment:
Even thought both String and Symbol are supported, I think it would be better to specify columns names with symbols, since that is what is being used to access the row values later on in the code example:

```ruby
results = db.read "users", [:id, name], keys: 1..5
```